crates/pallets/starknet/src/lib.rs

```diff
@@ -808,10 +810,11 @@ impl<T: Config> Pallet<T> {
     fn get_block_context() -> BlockContext {
         let block_number = UniqueSaturatedInto::<u64>::unique_saturated_into(frame_system::Pallet::<T>::block_number());
         let block_timestamp = Self::block_timestamp();
-        // Get fee token address. Its value is checked when we set it so it's fine to unwrap
-        let fee_token_address =
-            ContractAddress::try_from(StarkFelt::new(Self::fee_token_address().into()).unwrap()).unwrap();
         let chain_id = Self::chain_id_str();
+        let provided_fee_token_addr = Into::<StarkFelt>::into(Self::fee_token_address());
```

Comment:
this would be clearer
```suggestion
        let provided_fee_token_addr: StarkFelt = Self::fee_token_address().into();
```